source/grid/grid_in.cc

```diff
@@ -336,143 +336,183 @@ GridIn<dim, spacedim>::read_vtk(std::istream &in)
                    Utilities::int_to_string(subcelldata.boundary_lines.size()) +
                    ") in 2d."));
 
-      int tmp_int;
-      for (unsigned int i = 0; i < n_ints; ++i)
-        in >> tmp_int;
+        int tmp_int;
+        for (unsigned int i = 0; i < n_ints; ++i)
+            in >> tmp_int;
 
-      // Ignore everything up to CELL_DATA
-      while (in >> keyword)
-        if (keyword == "CELL_DATA")
-          {
-            unsigned int n_ids;
-            in >> n_ids;
-
-            AssertThrow(n_ids == n_geometric_objects,
-                        ExcMessage("The VTK reader found a CELL_DATA statement "
-                                   "that lists a total of " +
-                                   Utilities::int_to_string(n_ids) +
-                                   " cell data objects, but this needs to "
-                                   "equal the number of cells (which is " +
-                                   Utilities::int_to_string(cells.size()) +
-                                   ") plus the number of quads (" +
-                                   Utilities::int_to_string(
-                                     subcelldata.boundary_quads.size()) +
-                                   " in 3d or the number of lines (" +
-                                   Utilities::int_to_string(
-                                     subcelldata.boundary_lines.size()) +
-                                   ") in 2d."));
-
-            const std::vector<std::string> data_sets{"MaterialID",
-                                                     "ManifoldID"};
-
-            for (unsigned int i = 0; i < data_sets.size(); ++i)
-              {
-                // Ignore everything until we get to a SCALARS data set
-                while (in >> keyword)
-                  if (keyword == "SCALARS")
-                    {
-                      // Now see if we know about this type of data set,
-                      // if not, just ignore everything till the next SCALARS
-                      // keyword
-                      std::string set = "";
-                      in >> keyword;
-                      for (const auto &set_cmp : data_sets)
-                        if (keyword == set_cmp)
-                          {
-                            set = keyword;
-                            break;
-                          }
-                      if (set.empty())
-                        // keep ignoring everything until the next SCALARS
+        /////////////////////Processing the CELL_DATA and FIELD_DATA
+        /// section////////////////////////
```

Comment:
This comment looks a bit funnily formatted. Please break it into something more easily readable.